File: FIL/FIL.Web.Admin/ClientApp/models/CreateEventV1/TicketViewModel.ts
import { DiscountValueType } from '../../Enums/DiscountType';

export class TicketModel {
  etdId: number;
  name: string;
  description: string;
  ticketCategoryId: number;
  ticketCategoryNotes?: string;
  currencyId: number;
  currencyCode: string;
  quantity: number;
  isEnabled: boolean;
  price: number;
  ticketCategoryTypeId?: number;
  totalQuantity?: number;
  remainingQuantity?: number;
  ticketAltId?: string;
  promoCode?: string;
  discountValueType?: DiscountValueType;
  discountAmount?: number;
  discountPercentage?: number;
  isDiscountEnable?: boolean;
  donationAmount1?: number;
  donationAmount2?: number;
  donationAmount3?: number;
}

export class TicketViewModel {
  tickets: TicketModel[];
  eventId: number;
  eventDetailId?: number;
  isCreate?: boolean;
  success: boolean;
  currentStep?: number;
  isDraft?: boolean;
  completedStep?: string;
}

File: FIL/FIL.Web.Admin/ViewModels/CreateEventV1/TicketViewModel.cs
using System.Collections.Generic;
using System.IO;

namespace FIL.Web.Kitms.Feel.ViewModels.CreateEventV1
{
  public class TicketViewModel
  {
    public long EventId { get; set; }
    public long EventDetailId { get; set; }
    public bool IsCreate { get; set; }
    public bool Success { get; set; }
    public bool IsValidLink { get; set; }
    public bool IsDraft { get; set; }
    public string CompletedStep { get; set; }
    public short CurrentStep { get; set; }
    public List<FIL.Contracts.Models.CreateEventV1.TicketModel> Tickets { get; set; }
  }
}
